phoenix-core/src/main/java/org/apache/phoenix/schema/stats/UpdateStatisticsTool.java

```diff
@@ -164,12 +168,36 @@ void parseArgs(String[] args) {
         if (restoreDirOptionValue == null) {
             restoreDirOptionValue = getConf().get(FS_DEFAULT_NAME_KEY) + "/tmp";
         }
-        
+
+        jobPriority = getJobPriority(cmdLine);
+
         restoreDir = new Path(restoreDirOptionValue);
         manageSnapshot = cmdLine.hasOption(MANAGE_SNAPSHOT_OPTION.getOpt());
         isForeground = cmdLine.hasOption(RUN_FOREGROUND_OPTION.getOpt());
     }
 
+    public String getJobPriority() {
+        return this.jobPriority.toString();
+    }
+
+    private JobPriority getJobPriority(CommandLine cmdLine) {
+        String jobPriorityOption = cmdLine.getOptionValue(JOB_PRIORITY_OPTION.getOpt());
+         if (jobPriorityOption == null) {
+             return JobPriority.DEFAULT;
+         }
+
+         switch (jobPriorityOption) {
+             case "0" : return JobPriority.VERY_HIGH;
+             case "1" : return JobPriority.HIGH;
+             case "2" : return JobPriority.NORMAL;
+             case "3" : return JobPriority.LOW;
+             case "4" : return JobPriority.VERY_LOW;
+             case "5" : return JobPriority.DEFAULT;
```

Comment:
What is the `JobPriority.DEFAULT` level? Perhaps this should be the default case?

Reply:
https://issues.apache.org/jira/browse/MAPREDUCE-5870

Yes, if priority is not indicated by the user, it sets to default.

Reply:
maybe I should change to normal or default instead UNDEFINED_PRIORITY